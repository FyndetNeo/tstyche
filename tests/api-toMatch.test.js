import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toMatch");

test("'toMatch' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toMatch", expect.any(Function));
});

test("toMatch", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(stderr).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
