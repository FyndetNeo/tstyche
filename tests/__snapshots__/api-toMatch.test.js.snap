// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatch: stderr 1`] = `
"Error: Type '{ b: number; }' is not a subtype of type '{ a: string; b?: number | undefined; }'.

  44 |     expect<{ a: string; b?: number }>().type.toMatch<{ b?: number }>();
  45 | 
> 46 |     expect<{ a: string; b?: number }>().type.toMatch<{ b: number }>();
     |                                              ^
  47 |   });
  48 | 
  49 |   test("does NOT match expected type", () => {

       at ./__typetests__/toMatch.tst.ts:46:46 ❭ received type ❭ matches expected type

Error: Type '{ a: string; b?: number | undefined; }' is not a subtype of type '{ a: string; }'.

  53 |     expect<{ a: string }>().type.not.toMatch<{ a: string; b?: number }>();
  54 | 
> 55 |     expect<{ a: string }>().type.toMatch<{ a: string; b?: number }>();
     |                                  ^
  56 |   });
  57 | 
  58 |   test("matches expected value", () => {

       at ./__typetests__/toMatch.tst.ts:55:34 ❭ received type ❭ does NOT match expected type

Error: Type 'Names' is not a subtype of type '{ first: string; }'.

  60 |     expect<{ first: string; last?: string }>().type.toMatch(getNames());
  61 | 
> 62 |     expect<{ first: string }>().type.toMatch(getNames());
     |                                      ^
  63 |   });
  64 | 
  65 |   test("does NOT match expected value", () => {

       at ./__typetests__/toMatch.tst.ts:62:38 ❭ received type ❭ matches expected value

Error: Type 'Names' is not a subtype of type '{ first?: string | undefined; }'.

  66 |     expect<{ first?: string }>().type.not.toMatch(getNames());
  67 | 
> 68 |     expect<{ first?: string }>().type.toMatch(getNames());
     |                                       ^
  69 |   });
  70 | });
  71 | 

       at ./__typetests__/toMatch.tst.ts:68:39 ❭ received type ❭ does NOT match expected value

Error: Type '{ last: string; }' is not a subtype of type 'Names'.

  79 |     expect(getNames()).type.toMatch<{ last?: string }>();
  80 | 
> 81 |     expect(getNames()).type.toMatch<{ last: string }>();
     |                             ^
  82 |   });
  83 | 
  84 |   test("does NOT match expected type", () => {

       at ./__typetests__/toMatch.tst.ts:81:29 ❭ received value ❭ matches expected type

Error: Type '{ last: string; }' is not a subtype of type 'Names'.

  86 | 
  87 |     expect(getNames()).type.not.toMatch<{ last: string }>();
> 88 |     expect(getNames()).type.toMatch<{ last: string }>();
     |                             ^
  89 |   });
  90 | 
  91 |   test("matches expected value", () => {

       at ./__typetests__/toMatch.tst.ts:88:29 ❭ received value ❭ does NOT match expected type

Error: Type 'Names' is not a subtype of type '{ last: string; }'.

   95 |     expect({ first: "One", last: "Two" }).type.toMatch(getNames());
   96 | 
>  97 |     expect({ last: "Two" }).type.toMatch(getNames());
      |                                  ^
   98 |   });
   99 | 
  100 |   test("does NOT match expected value", () => {

        at ./__typetests__/toMatch.tst.ts:97:34 ❭ received value ❭ matches expected value

Error: Type 'Names' is a subtype of type '{ first: string; }'.

  101 |     expect({ last: "Two" }).type.not.toMatch(getNames());
  102 | 
> 103 |     expect({ first: "One" }).type.not.toMatch(getNames());
      |                                       ^
  104 |   });
  105 | });
  106 | 

        at ./__typetests__/toMatch.tst.ts:103:39 ❭ received value ❭ does NOT match expected value

"
`;

exports[`toMatch: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toMatch.tst.ts
  + difference from '.toBeAssignable()'
  received type
    × matches expected type
    × does NOT match expected type
    × matches expected value
    × does NOT match expected value
  received value
    × matches expected type
    × does NOT match expected type
    × matches expected value
    × does NOT match expected value

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      8 failed, 1 passed, 9 total
Assertions: 8 failed, 28 passed, 36 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
